docs/aws/audit/lambdamonitoring/rules/lambda_concurrency_remediation.mdx

```diff
@@ -1,6 +1,201 @@
+
+   response = set_lambda_concurrency(function_name, reserved_concurrent_executions)
+   print(response)
+   ```
+
+4. **Run the Script:**
+   Execute the Python script to apply the concurrency settings to your Lambda function.
+
+   ```bash
+   python set_lambda_concurrency.py
+   ```
+
+By following these steps, you can ensure that your AWS Lambda functions have concurrency configured, preventing potential misconfigurations related to unbounded concurrency.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the AWS Lambda console.
+
+2. In the navigation pane, choose "Functions".
+
+3. Select the function you want to check for concurrency configuration.
+
+4. In the function's detail page, scroll down to the "Concurrency" section. If there is no concurrency limit set, it means the function is using unreserved concurrency which could potentially lead to throttling issues.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Lambda functions.
+
+2. Once the AWS CLI is installed and configured, you can list all the Lambda functions in your account by running the following command:
+
+   ```
+   aws lambda list-functions --region your-region-name
+   ```
+
+   Replace 'your-region-name' with the name of the AWS region where your Lambda functions are deployed. This command will return a JSON object with details about all your Lambda functions.
+
+3. To check the concurrency configuration of a specific Lambda function, you can use the following command:
+
+   ```
+   aws lambda get-function --function-name your-function-name --region your-region-name
+   ```
+
+   Replace 'your-function-name' with the name of the Lambda function you want to check, and 'your-region-name' with the name of the AWS region where the function is deployed. This command will return a JSON object with details about the specified Lambda function.
+
+4. In the returned JSON object, look for the 'Concurrency' field. If this field is not present or if its value is not set, then the Lambda function does not have a concurrency configuration. If the 'Concurrency' field is present and its value is set, then the Lambda function has a concurrency configuration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. First, you need to install the AWS SDK for Python (Boto3) in your environment. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary libraries and establish a session with AWS. You will need your AWS access key and secret access key for this.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>